WordPress/gutenberg
# HorizontalRule component for cross-platform separator block

## Description

This PR adds a `HorizontalRule` wrapper to the `hr` HTML tag, using [react-native-hr](https://github.com/Riglerr/react-native-hr) on the native implementation, to allow for a cross platform  separator block.

## How has this been tested?




Tested on mobile:

- Create a new post
- Insert a new block, verify that separator is available
- Insert a separator, verify it renders correctly

Tested on web:

- Create a new post
- Insert a separator block, verify it renders correctly, block styles still work
- Preview the post and verify it renders correctly

## Screenshots 

![Simulator Screen Shot - iPhone Xs - 2019-04-17 at 10 16 46](https://user-images.githubusercontent.com/8739/56271949-f11f6300-60f9-11e9-99a8-76f696b63602.png)


## Types of changes





## Checklist:
